gp-includes/things/note.php

```diff
@@ -0,0 +1,138 @@
+<?php
+/**
+ * Things: GP_Note class
+ *
+ * @package GlotPress
+ * @subpackage Things
+ * @since 3.0.0
+ */
+
+/**
+ * Core class used to implement the notes system.
+ *
+ * @since 3.0.0
+ */
+class GP_Note extends GP_Thing {
+
+	/**
+	 * Name of the database table.
+	 *
+	 * @var string $table_basename
+	 */
+	public $table_basename = 'gp_notes';
+
+	/**
+	 * List of field names for a translation.
+	 *
+	 * @var array $field_names
+	 */
+	public $field_names = array(
+		'id',
+		'original_id',
+		'translation_set_id',
+		'note',
+		'user_id',
+		'date_added',
+		'date_modified',
+	);
+
+	/**
+	 * List of field names which have an integer value.
+	 *
+	 * @var array $int_fields
+	 */
+	public $int_fields = array(
+		'id',
+		'original_id',
+		'translation_set_id',
+		'user_id',
+	);
+
+	/**
+	 * List of field names which cannot be updated.
+	 *
+	 * @var array $non_updatable_attributes
+	 */
+	public $non_updatable_attributes = array( 'id' );
+
+	/**
+	 * SQL string for order by date
+	 *
+	 * @var array $default_order
+	 */
+	public $default_order = 'ORDER BY date_added DESC';
+
+	/**
+	 * Save the note
+	 *
+	 * @since 3.0.0
+	 *
+	 * @param string $args    Parameters that are not used.
+	 *
+	 * @return object The output of the query.
+	 */
+	public function save( $args = null ) {
+		global $wpdb;
+		$original_id    = gp_post( 'original_id' );
+		$translation_id = gp_post( 'translation_id' );
+		$note           = trim( gp_post( 'note' ) );
```

Comment:
Before the note gets saved in the database it needs to be sanitized. I'd probably handling them the same as core does with comments.